pkg/abstractions/endpoint/endpoint.go

```diff
@@ -212,10 +213,14 @@ func (es *HttpEndpointService) forwardRequest(
 }
 
 func (es *HttpEndpointService) InstanceFactory(stubId string, options ...func(abstractions.IAutoscaledInstance)) (abstractions.IAutoscaledInstance, error) {
-	return es.getOrCreateEndpointInstance(stubId)
+	return es.getOrCreateEndpointInstance(es.ctx, stubId)
 }
 
-func (es *HttpEndpointService) getOrCreateEndpointInstance(stubId string, options ...func(*endpointInstance)) (*endpointInstance, error) {
+func (es *HttpEndpointService) getOrCreateEndpointInstance(ctx context.Context, stubId string, options ...func(*endpointInstance)) (*endpointInstance, error) {
+	trace := common.TraceFunc(ctx, "pkg/abstractions/endpoint", "HttpEndpointService.getOrCreateEndpointInstance", es.config.DebugMode,
```

Comment:
We should start using the trace.ctx in the following functions to make things consistent.

Reply:
```
stub, err := es.backendRepo.GetStubByExternalId(trace.ctx, stubId)
	if err != nil {
		return nil, errors.New("invalid stub id")
	}
```